src/core/tensor/tensor.cc

```diff
@@ -272,19 +310,27 @@ void Tensor::ToProto(singa::TensorProto *proto) const {
 Tensor Tensor::Clone(std::shared_ptr<Device> device) const {
   if (device == nullptr) device = device_;
   Tensor t(shape_, device_, data_type_);
-  t.transpose_ = transpose_;
+  //t.transpose_ = transpose_;
+  t.strides_ = strides_;
   t.CopyData(*this);
   return t;
 }
 
+//yisen todo
+//clarify transpose vs copy
```

Comment:
transpose does not copy data.
a=b.T() --> a and b share the same memory.
a=b.Copy() --> a and b does not share memory.